docs/source/development/automated_testing.md

```diff
@@ -0,0 +1,165 @@
+
+Let's look at how you can start working with `pytest` in your Kedro project.
+
+### Install `pytest`
+
+Install `pytest` as you would install other packages with `pip`, making sure your project's virtual environment is active.
+
+```bash
+pip install pytest
+```
+
+### Create a `/tests` directory
+
+Now that `pytest` is installed, you will need a place to put your tests. Create a `/tests` folder in the `/src` directory of your project.
+
+```bash
+mkdir <project_root>/src/tests
+```
+
+### Test directory structure
+
+The subdirectories in your project's `/tests` directory should mirror the directory structure of your project's `/src/<project_name>` directory. All files in the `/tests` folder should be named `test_<file_being_tested>.py`. For instance:
+
+```
+src
+│   ...
+└───<project_name>
+│   └───pipelines
+│       └───dataprocessing
+│           │   ...
+│           │   nodes.py
+│           │   ...
+│   
+└───tests
+│   └───pipelines
+│       └───dataprocessing
+│           │   ...
+│           │   test_nodes.py
+│           │   ...
+```
+
+### Create an example test
+
+Now that you have a place to put your tests, you can create an example test in the new file `/src/tests/test_run.py`. The example test simply checks that the project_path attribute of a specially-defined `KedroContext` object has been correctly set.
+
+```
+import pytest
+from kedro.config import ConfigLoader
+from kedro.framework.context import KedroContext
+from kedro.framework.hooks import _create_hook_manager
+
+
+@pytest.fixture
+def config_loader():
+    return ConfigLoader(conf_source=str(Path.cwd()))
+
+
+@pytest.fixture
+def project_context(config_loader):
+    return KedroContext(
+        package_name=<package_name>,
+        project_path=Path.cwd(),
+        config_loader=config_loader,
+        hook_manager=_create_hook_manager(),
+    )
+
+class TestProjectContext:
+    def test_project_path(self, project_context):
+        assert project_context.project_path == Path.cwd()
+```
+
+This test is redundant, but it introduces a few of `pytest`'s core features and demonstrates the layout of a test file:
+- [Fixtures](https://docs.pytest.org/en/7.1.x/explanation/fixtures.html#about-fixtures) are used to define resources used in tests.
+- Tests are implemented in methods or functions beginning with `test_` and classes beginning with `Test`.
+- The `assert` statement is used to compare the result of the test with an expected value.
+
+You can read more about the basics of using `pytest` [here](https://docs.pytest.org/en/7.1.x/getting-started.html). For help writing your own tests and using all of the features of `pytest`, see the [docs](https://docs.pytest.org/).
+
+### Run your tests
+
+To run your tests, run `pytest` from within your project's root directory or from the `/src` directory.
```

Comment:
Maybe we don't need need to mention the `src/`. In theory, `pytest` would still work, but the following section we also mention `pyproject.toml`, and it won't be picked up if it's running from `src/`